src/freezing/freezing_apply_changes.py

```diff
@@ -1,156 +1,292 @@
+
+    Returns:
+        amended_df (pd.DataFrame): The main snapshot with amendments applied.
+    """
+    if not validate_amendments_df(main_df, amendments_df, freezing_logger):
+        freezing_logger.info("Skipping amendments since the amendments csv is invalid...")
+        return main_df
+
+    changes_refs = amendments_df[
+        amendments_df.accept_changes==True
+    ].reference.unique()
+
+    accepted_amendments_df = amendments_df[
+        amendments_df.reference.isin(changes_refs)
+    ]
+
+    if accepted_amendments_df.shape[0] == 0:
+        freezing_logger.info(
+            "Amendments file contained no records marked for inclusion"
+        )
+        return main_df
+
+    # Drop the diff columns and accept_changes col
+    accepted_amendments_df = accepted_amendments_df.drop(
+        columns=[col for col in accepted_amendments_df.columns if col.endswith("_diff")]
+    )
+    accepted_amendments_df = accepted_amendments_df.drop("accept_changes", axis=1)
+
+    #rename columns
+    accepted_amendments_df.columns = [
+        col.replace("_updated", "") for col in accepted_amendments_df.columns
+    ]
+
+    # update last_frozen column
+    accepted_amendments_df = _add_last_frozen_column(accepted_amendments_df, run_id)
+
+    # drop records to be amended from main df
+    main_df = main_df[~main_df.reference.isin(changes_refs)]
+
+    # add amended records to main df
+    amended_df = pd.concat([main_df, accepted_amendments_df])
+
+    freezing_logger.info(
+        f"{accepted_amendments_df.shape[0]} records amended during freezing"
+    )
+    return amended_df
+
+
+def apply_additions(
+    main_df: pd.DataFrame,
+    additions_df: pd.DataFrame,
+    run_id: int,
+    freezing_logger: logging.Logger
+    ) -> pd.DataFrame:
+    """Apply additions to the main snapshot.
+
+    Args:
+        main_df (pd.DataFrame): The main snapshot.
+        additions_df (pd.DataFrame): The additions to apply.
+        run_id (int): The current run id.
+        freezing_logger (logging.Logger): The logger.
+
+    Returns:
+        added_df (pd.DataFrame): The main snapshot with additions applied.
+    """
+    if not validate_additions_df(main_df, additions_df, freezing_logger):
+        freezing_logger("Skipping additions since the additions csv is invalid...")
+        return main_df
+    # Drop records where accept_changes is False and if any remain, add them to main df
+    changes_refs = additions_df[
+        additions_df.accept_changes==True
+    ].reference.unique()
+
+    accepted_additions_df = additions_df[
+        additions_df.reference.isin(changes_refs)
+    ]
+    if accepted_additions_df.shape[0] > 0:
+        accepted_additions_df = _add_last_frozen_column(accepted_additions_df, run_id)
+        added_df = pd.concat([main_df, accepted_additions_df], ignore_index=True)
+        freezing_logger.info(
+            f"{accepted_additions_df.shape[0]} records added during freezing"
+        )
```

Comment:
The accept_changes col needs dropping for additions (it's already dropped for amends)